conda_gitenv/deploy.py

```diff
@@ -125,50 +161,78 @@ def deploy_repo(repo, target, desired_env_labels=None):
             # skip this environment.
             if manifest_branch_name not in repo.branches:
                 continue
-            manifest_branch = repo.branches[manifest_branch_name]
             branch.checkout()
-            all_labelled_tags = tags_by_label(os.path.join(repo.working_dir, 'labels'))
+            all_labelled_tags = tags_by_label(os.path.join(repo.working_dir,
+                                                           'labels'))
 
-            # Create a latest tag that points to the most recently tagged environment.
+            # Create a latest tag that points to the most recently tagged
+            # environment.
             if env_tags.get(branch.name):
                 latest_tag = max(env_tags[branch.name],
                                  key=lambda t: t.commit.committed_date)
                 all_labelled_tags['latest'] = latest_tag.name
 
             # Only deploy environments that match the given pattern.
             labelled_tags = {}
-            if desired_env_labels is None:
-                desired_env_labels = ['*']
+            if env_labels is None:
+                env_labels = ['*']
             for label, tag in all_labelled_tags.items():
-                if any([fnmatch.fnmatch('{}/{}'.format(branch.name, label),
-                                        env_label) for env_label in desired_env_labels]):
+                item = '{}/{}'.format(branch.name, label)
+                match = [fnmatch(item, env_label)
+                         for env_label in env_labels]
+                if any(match):
                     labelled_tags[label] = tag
 
             for tag in set(labelled_tags.values()):
-                deploy_tag(repo, tag, target)
+                deploy_tag(repo, tag, target,
+                           api_user=api_user, api_key=api_key)
+
+            # Lock down the package cache files which may contain
+            # API credentials.
+            mode = stat.S_IRUSR | stat.S_IWUSR
+            pkg_cache_urls = os.path.join(target, PKG_CACHE_NAME, 'urls.txt')
```

Comment:
We could always strip the API stuff from the urls.txt file at the end?

Reply:
This is done by `conda` itself. Not sure of the impact if we did strip out the API details, whether it would break something or not.

For a happy compromise, how I about I keep this as is, but create an issue to investigate?